x-pack/plugins/security_solution/public/timelines/components/timeline/tabs/query/header/index.tsx

```diff
@@ -74,41 +90,69 @@ const QueryTabHeaderComponent: React.FC<Props> = ({
+    <StyledEuiFlyoutHeader data-test-subj={`${activeTab}-tab-flyout-header`} hasBorder={false}>
+      <InPortal node={timelineEventsCountPortalNode}>
+        {showEventsCountBadge ? <EventsCountBadge>{totalCount}</EventsCountBadge> : null}
+      </InPortal>
+      <EuiFlexGroup gutterSize="s" direction="column">
+        {!unifiedComponentsInTimelineEnabled &&
+          timelineFullScreen &&
+          setTimelineFullScreen != null && (
+            <EuiFlexItem>
+              <EuiFlexGroup alignItems="center" gutterSize="s">
+                <ExitFullScreen
+                  fullScreen={timelineFullScreen}
+                  setFullScreen={setTimelineFullScreen}
+                />
+              </EuiFlexGroup>
+            </EuiFlexItem>
+          )}
+        <EuiFlexItem data-test-subj="timeline-date-picker-container">
+          <TabHeaderContainer data-test-subj="timelineHeader">
+            <EuiFlexGroup gutterSize="s" direction="column">
+              <EuiFlexItem>
+                <StatefulSearchOrFilter filterManager={filterManager} timelineId={timelineId} />
+              </EuiFlexItem>
+              {showCallOutUnauthorizedMsg && (
+                <EuiFlexItem>
+                  <EuiCallOut
+                    data-test-subj="timelineCallOutUnauthorized"
+                    title={i18n.CALL_OUT_UNAUTHORIZED_MSG}
+                    color="warning"
+                    iconType="warning"
+                    size="s"
+                  />
+                </EuiFlexItem>
+              )}
+              {status === TimelineStatus.immutable && (
+                <EuiFlexItem>
+                  <EuiCallOut
+                    data-test-subj="timelineImmutableCallOut"
+                    title={i18n.CALL_OUT_IMMUTABLE}
+                    color="primary"
+                    iconType="warning"
+                    size="s"
+                  />
+                </EuiFlexItem>
+              )}
+              {show ? (
+                <DataProvidersContainer
+                  className="data-providers-container"
+                  $shouldShowQueryBuilder={shouldShowQueryBuilder}
+                >
+                  <DataProviders timelineId={timelineId} />
+                </DataProvidersContainer>
+              ) : null}
+            </EuiFlexGroup>
+          </TabHeaderContainer>
         </EuiFlexItem>
-      )}
-      {show ? (
-        <DataProvidersContainer
-          className="data-providers-container"
-          $shouldShowQueryBuilder={shouldShowQueryBuilder}
-        >
-          <DataProviders timelineId={timelineId} />
-        </DataProvidersContainer>
-      ) : null}
-    </EuiFlexGroup>
+        {/* TODO: This is a temporary solution to hide the KPIs until lens components play nicely with timelines */}
+        {/* https://github.com/elastic/kibana/issues/17156 */}
+        {/* <EuiFlexItem grow={false}> */}
+        {/*   <TimelineKpi timelineId={timelineId} /> */}
+        {/* </EuiFlexItem> */}
```

Comment:
can we please remove this code? It's been sitting there for months (years?) now 😆 

If we ever want to bring it back it will be in the git history...

Reply:
We can, and only like 3 months or so 😂 . Jatin commented it out with the timeline query tab refactor iirc